packages/kit/src/runtime/app/images.js

```diff
@@ -0,0 +1,122 @@
+import { DEV } from 'esm-env';
+import { widths as default_widths, loader } from '__sveltekit/images';
+
+/**
+ * @typedef {{
+ *  src: string;
+ *  sizes?: string;
+ *  fetchpriority?: 'high' | 'low' | 'auto';
+ *  loading?: 'eager' | 'lazy';
+ *  style?: string;
+ *  class?: string;
+ *  alt?: string;
+ *  options?: Record<string, any>;
+ *  } & ({ width: number; height: number } | { layout: 'fill' | 'custom', width?: number; height?: number })
+ * } ImageOptions
+ */
+
+/**
+ * @param {ImageOptions} payload
+ * @returns {{ src: string, srcset: string, sizes?: string, width?: number, height?: number, loading?: 'eager' | 'lazy', fetchpriority?: 'high' | 'low' | 'auto', style?: string, class?: string, alt?: string }}
+ */
+export function getImage(payload) {
+	const src = payload.src;
+	const widths = get_widths(payload.width, payload.sizes);
+
+	if (DEV) {
+		if (!('width' in payload) && !('layout' in payload)) {
+			console.warn(
+				`Missing 'width' or 'layout' for image ${src}. Provide one of them to prevent layout shift.`
+			);
+		}
+	}
+
+	let _src = src;
+	const srcset = widths.widths
+		.map((width, i) => {
+			const url = DEV ? src : loader(src, width, payload.options);
+			_src = url; // ensures that the largest one is set
+			const w = widths.kind === 'x' ? `${i + 1}x` : `${width}w`;
+			return `${url} ${w}`;
+		})
+		.join(', ');
+
+	let style = payload.style;
+	// @ts-expect-error
+	if (payload.layout === 'fill') {
+		style = 'position:absolute;width:100%;height:100%;inset:0px;' + (style || '');
+	}
+
+	// Order of attributes is important here as they are set in this order
+	// and having src before srcset would result in a flicker
```

Comment:
Better safe than sorry, but I haven't noticed any flicker when `src` comes before `srcset` in an `img` tag. Just curious if you have any proof of this? Haven't seen it mentioned anywhere else.